### ISSUE TITLE
Cold start optimization: investigate prefetching in fetcher

### ISSUE THREAD

**Issue by soamvasani** [fission/fission#205]
We could have fetcher prefetch and uncompress functions. This will be more noticeable when functions are in a several-MB package that needs to be unzipped. We'll have to measure how much this actually buys us.

**Comment by ajbouh** [comment]
I've experimented with adding an http proxy daemonset to all nodes that will run functions. This sped up "warm fetching" to ~200MB/s. Any interest in doing something like this by default?

**Comment by soamvasani** [comment]
@ajbouh Yes, we very interested! Could you share a bit more about what you did, the benchmark, etc.?

**Comment by ajbouh** [comment]
My experiment was comparing fetching a 100MB file from a cdn directly it
from that cdn via a squid proxy in a daemonset.

I tested via wget -O > /dev/null in a session attached to a container
running on the node. This was on a modest machine on gke.

This is a gist of the k8s resources I used to set things up and test.
https://gist.github.com/ajbouh/1c2d21b0d062a11d7944e6eaf3967606
